include/zephyr/irq.h

```diff
@@ -291,10 +300,11 @@ static inline unsigned int irq_get_level(unsigned int irq)
  */
 static inline unsigned int irq_from_level_2(unsigned int irq)
 {
-#ifdef CONFIG_3RD_LEVEL_INTERRUPTS
-	return ((irq >> 8) & 0xFF) - 1;
+#if defined(CONFIG_3RD_LEVEL_INTERRUPTS)
+		return ((irq >> CONFIG_2ND_LEVEL_INTERRUPT_BITS) &
+			BIT_MASK(CONFIG_1ST_LEVEL_INTERRUPT_BITS)) - 1;
```

Comment:
is this function trying to extract just the 2nd level interrupt?

```suggestion
		return ((irq >> CONFIG_1ST_LEVEL_INTERRUPT_BITS) &
			BIT_MASK(CONFIG_2ND_LEVEL_INTERRUPT_BITS)) - 1;
```

Reply:
I think this should actually be the sum of the first and second levels in order to get to the third level bits, but good catch! Thank you!